yt_dlp/extractor/applepodcasts.py

```diff
@@ -39,47 +51,25 @@ class ApplePodcastsIE(InfoExtractor):
     def _real_extract(self, url):
         episode_id = self._match_id(url)
         webpage = self._download_webpage(url, episode_id)
-        episode_data = {}
-        ember_data = {}
-        # new page type 2021-11
-        amp_data = self._parse_json(self._search_regex(
-            r'(?s)id="shoebox-media-api-cache-amp-podcasts"[^>]*>\s*({.+?})\s*<',
-            webpage, 'AMP data', default='{}'), episode_id, fatal=False) or {}
-        amp_data = try_get(amp_data,
-                           lambda a: self._parse_json(
-                               next(a[x] for x in iter(a) if episode_id in x),
-                               episode_id),
-                           dict) or {}
-        amp_data = amp_data.get('d') or []
-        episode_data = try_get(
-            amp_data,
-            lambda a: next(x for x in a
-                           if x['type'] == 'podcast-episodes' and x['id'] == episode_id),
-            dict)
-        if not episode_data:
-            # try pre 2021-11 page type: TODO: consider deleting if no longer used
-            ember_data = self._parse_json(self._search_regex(
-                r'(?s)id="shoebox-ember-data-store"[^>]*>\s*({.+?})\s*<',
-                webpage, 'ember data'), episode_id) or {}
-            ember_data = ember_data.get(episode_id) or ember_data
-            episode_data = try_get(ember_data, lambda x: x['data'], dict)
-        episode = episode_data['attributes']
-        description = episode.get('description') or {}
-
-        series = None
-        for inc in (amp_data or ember_data.get('included') or []):
-            if inc.get('type') == 'media/podcast':
-                series = try_get(inc, lambda x: x['attributes']['name'])
-        series = series or clean_html(get_element_by_class('podcast-header__identity', webpage))
+        server_data = self._parse_json(
+            get_element_by_id('serialized-server-data', webpage),
+            episode_id) or [{}]
+        model_data = traverse_obj(
+            server_data,
+            (0, 'data', 'headerButtonItems',
+             {lambda x: next(y for y in x
+                             if y.get('$kind') == 'bookmark' and y.get('modelType') == 'EpisodeOffer')},
+             'model'))
```

Comment:
Let's simplify this to:
```suggestion
        model_data = traverse_obj(server_data, (
            'headerButtonItems', lambda _, v: v['$kind'] == 'bookmark' and v['modelType'] == 'EpisodeOffer',
            'model', {dict}, any))
```
Also, was there any reason for choosing `bookmark`/`EpisodeOffer`? `download`/`EpisodeLockup` seems more fitting and has more metadata available.

Reply:
i chose `bookmark`/`EpisodeOffer`, over `download`/`EpisodeLockup`, because while all other attributes are identical between the two (ie `.model.title`, `.model.releaseDate`, & `.model.duration`), url is a top-level attribute in `bookmark`/`EpisodeOffer` compared to deeply nested in `download`/`EpisodeLockup` (ie `.model.streamUrl` vs `.model.playAction.episodeOffer.streamUrl`, respectively).

yes, `download`/`EpisodeLockup` has more metadata, including a copy of `bookmark`/`EpisodeOffer` (ie `.model.playAction.episodeOffer`), but if:
1. all the wanted attributes are in `bookmark`/`EpisodeOffer`
2. the url in `download`/`EpisodeLockup` is in an embedded copy of `bookmark`/`EpisodeOffer`

then why not just use `bookmark`/`EpisodeOffer`?

Reply:
Alright then, let's keep `bookmark`/`EpisodeOffer`. Theoretically, we could add a fallback, but I don't think it's worth it.